app/controllers/plans_controller.rb

```
@@ -29,12 +31,26 @@ def create
       flash[:alert] = %(
         #{count} #{'error'.pluralize(count)}
         prohibited this plan from being saved:
-        #{ @plan.errors.full_messages.join('. ')}
+        #{@plan.errors.full_messages.join('. ')}
       )
       render :new
     end
   end
 
+  def destroy
+    @plan = Plan.find(params[:id])
+    if @plan.destroy!
```

I would only do `#destroy`. This would be a silent error and the server wouldn't crash and disrupt the user experience.